ember_debug/route-debug.js

```diff
@@ -168,6 +174,12 @@ function buildSubTree(routeTree, route) {
         // Ember < 3.9.0
         routeHandler = routerLib.getHandler(handler);
       }
+
+      // Skip when route is an unresolved promise
+      if (typeof routeHandler?.then === 'function') {
+        continue;
```

Comment:
> buildSubtree keeps getting called until all the promises are settled anyway

`buildSubTree` is definitely called many times (once for each route),

I think we should do something like:

```js
      // Skip when route is an unresolved promise
      if (typeof routeHandler?.then === 'function') {
        // ensure we rebuild the route tree when this route is resolved
        routeHandler.then(() => this.notifyPropertyChange('routeTree'));
        controllerName = '(unresolved)';
        controllerClassName = '(unresolved)';
        templateName = '(unresolved)';
      } else {
        controllerName =
          routeHandler.get('controllerName') || routeHandler.get('routeName');
        controllerFactory = owner.factoryFor
          ? owner.factoryFor(`controller:${controllerName}`)
          : owner._lookupFactory(`controller:${controllerName}`);
        controllerClassName = this.getClassName(controllerName, 'controller');
        templateName = this.getClassName(handler, 'template');
      }

      subTree[handler] = {
        value: {
          name: handler,
          routeHandler: {
            className: routeClassName,
            name: handler,
          },
          controller: {
            className: controllerClassName,
            name: controllerName,
            exists: !!controllerFactory,
          },
          template: {
            name: templateName,
          },
        },
      };
```

This ensures that the subtree structure itself is correct (and does indicate that _something_ is at this location / route name), it also ensures that a re-computation will happen when the promise resolves.

Reply:
Thanks for this detailed suggestion @rwjblue! @steventsao would you be interested in updating the PR accordingly?

Reply:
Done @rwwagner90